src/utils/recommendationUtility.js

```diff
@@ -0,0 +1,40 @@
+const SpotifyWebApi = require('spotify-web-api-node'); // Import Spotify Web API library
+const AlbumRecommendation = require('../models/AlbumRecommendation');
+
+// Set up Spotify Web API client with credentials
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.SPOTIFY_CLIENT_ID,
+  clientSecret: process.env.SPOTIFY_CLIENT_SECRET,
+});
+
+async function generateRecommendations(userId, listeningHistory) {
+  try {
+    // Check if listeningHistory is empty
+    if (listeningHistory.length === 0) {
+      console.warn('No listening history found for user:', userId);
+      return;
+    }
+
+    // Extract artistIds from listeningHistory
+    const artistIds = listeningHistory.map((entry) => entry.artistId);
+
+    // Make a call to Spotify's "Get Recommendations" endpoint
```

Comment:
Before we can make any calls to Spotify's api we need to authenticate our session. We can follow the same pattern we've done over in the `fetchAlbumSpotify` file:

https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/321237cb8427951d64b61e7c73e1a6062f131ff3/fetchAlbumSpotify.js#L21-L23